vispy/visuals/graphs/util.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+"""
+Graph utilities
+===============
+
+A module containing several graph utility functions
+"""
+
+import itertools
+
+import numpy as np
+
+
+def straight_line_vertices(adjacency_mat, node_coords, directed=False):
+    if adjacency_mat.shape[0] != adjacency_mat.shape[1]:
+        raise ValueError("Adjacency matrix should be square.")
+
+    num_nodes = adjacency_mat.shape[0]
+
+    line_vertices = []
+    arrows = []
+    for edge in itertools.combinations(range(num_nodes), 2):
+        reverse = (edge[1], edge[0])
+
+        if adjacency_mat[edge] == 1 or adjacency_mat[reverse] == 1:
+            line_vertices.extend([node_coords[edge[0]],
+                                  node_coords[edge[1]]])
+
+        if directed:
+            if adjacency_mat[edge] == 1 and adjacency_mat[reverse] == 0:
+                arrows.extend([
+                    node_coords[edge[0]],
+                    node_coords[edge[1]]
+                ])
+            elif adjacency_mat[edge] == 0 and adjacency_mat[reverse] == 0:
+                arrows.extend([
+                    node_coords[reverse[0]],
+                    node_coords[reverse[1]]
+                ])
```

Comment:
We will definitely need to vectorize this at some point, it should be possible. But for now it's okay to be explicit :)